aws/data_source_aws_nat_gateway.go

```diff
@@ -116,6 +122,7 @@ func dataSourceAwsNatGatewayRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("state", ngw.State)
 	d.Set("subnet_id", ngw.SubnetId)
 	d.Set("vpc_id", ngw.VpcId)
+	d.Set("tags", tagsToMap(ngw.Tags))
```

Comment:
When using `d.Set()` with aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types). e.g.

```go
if err := d.Set("tags", tagstoMap(ngw.Tags)); err != nil {
  return fmt.Errorf("error setting tags: %s", err)
}
```

Since the `tagsToMap()` function has well established usage across other resources I won't treat this as a blocker here though. 👍